pytest_fixtures/component/provision_gce.py

```diff
@@ -5,40 +5,69 @@
 import pytest
 from airgun.session import Session
 from fauxfactory import gen_string
-from nailgun import entities
 from wrapanapi.systems.google import GoogleCloudSystem
 
 from robottelo.config import settings
+from robottelo.constants import DEFAULT_ARCHITECTURE
 from robottelo.constants import DEFAULT_PTABLE
 from robottelo.constants import FOREMAN_PROVIDERS
 from robottelo.exceptions import GCECertNotFoundError
 
 
 @pytest.fixture(scope='session')
-def gce_cert(session_target_sat):
-    _, gce_cert_file = mkstemp(suffix='.json')
-    cert = json.loads(settings.gce.cert)
-    cert['local_path'] = gce_cert_file
-    with open(gce_cert_file, 'w') as f:
-        json.dump(cert, f)
-    session_target_sat.put(gce_cert_file, settings.gce.cert_path)
-    if session_target_sat.execute(f'[ -f {settings.gce.cert_path} ]').status != 0:
-        raise GCECertNotFoundError(
-            f"The GCE certificate in path {settings.gce.cert_path} is not found in satellite."
-        )
-    return cert
+def sat_gce(request, session_puppet_enabled_sat, session_target_sat):
+    hosts = {'sat': session_target_sat, 'puppet_sat': session_puppet_enabled_sat}
+    yield hosts[request.param]
```

Comment:
I see, we've some tests which we marking only for sat, so would suggest using getattr where we make sat default and you don't need to mark parametrize if you don't want to run on both sat
```suggestion
    hosts = {'sat': session_target_sat, 'puppet_sat': session_puppet_enabled_sat}
    yield hosts[getattr(request, 'param', 'sat')]
```

Reply:
Works well @Gauravtalreja1